opentelemetry-sdk/src/trace/sampler/jaeger_remote/sampling_strategy.rs

```diff
@@ -107,9 +107,10 @@ impl Inner {
                 }
             })
             .unwrap_or_else(|_err| {
-                global::handle_error(TraceError::Other(
-                    "jaeger remote sampler mutex poisoned".into(),
-                ))
+                otel_debug!(
```

Comment:
what is the impact of this poisoning? is that self-recoverable? If this causes JaegerSampler to stop functioning, then it should be WARN severity?

Reply:
It is not self-recoverable. The application will not stop, however the Jaeger remote sampling logic will not be used. We can make this otel_warn, which will get logged by default every 5 minutes.

Reply:
We will revert to previous strateegy. User may see unexpected spans but it's propbably not critical. Debug seems to be fine to start with

Reply:
Thanks, keeping it otel_debug:
```
                otel_debug!(
                    name: "JaegerRemoteSampler.MutexPoisoned",
                    message = "Failed to update Jaeger Remote sampling strategy. The sampler's internal mutex is poisoned, indicating a panic occurred in another thread holding the lock. Sampling decisions may be using stale configuration.",
                );
```

Reply:
I think this should be WARN. Having expected spans (or viceversa) is something we want users to be aware of.

Reply:
```rust
 otel_warn!(
      name: "JaegerRemoteSampler.MutexPoisoned"
      message = "Unable to update Jaeger Remote sampling strategy: the sampler's internal mutex is 
poisoned, likely due to a panic in another thread holding the lock. The last known configuration will 
continue to be used until the remote sampling client is restarted.",
);
```